File: OsmAnd-java/src/main/java/net/osmand/search/core/CustomSearchPoiFilter.java
package net.osmand.search.core;

import net.osmand.ResultMatcher;
import net.osmand.binary.BinaryMapIndexReader.SearchPoiTypeFilter;
import net.osmand.data.Amenity;

public interface CustomSearchPoiFilter extends SearchPoiTypeFilter {

	public String getFilterId();

	public String getName();

	public Object getIconResource();

	public ResultMatcher<Amenity> wrapResultMatcher(final ResultMatcher<Amenity> matcher);

}

File: OsmAnd-shared/src/commonMain/kotlin/net/osmand/shared/search/core/CustomSearchPoiFilter.kt
package net.osmand.shared.search.core

import net.osmand.shared.binary.ResultMatcher
import net.osmand.shared.binary.SearchPoiTypeFilter
import net.osmand.shared.data.Amenity

/**
 * A poi filter of the app that the search offers as a result, such as a saved filter.
 *
 * A copy of `CustomSearchPoiFilter` in OsmAnd-java, which stays there for android and tools; this
 * copy is for iOS.
 */
interface CustomSearchPoiFilter : SearchPoiTypeFilter {

	fun getFilterId(): String?

	fun getName(): String?

	fun getIconResource(): Any?

	fun wrapResultMatcher(matcher: ResultMatcher<Amenity>): ResultMatcher<Amenity>

	fun getDefaultSearchType(): SearchSettings.SortType? = null
}
